hack/release.sh

```diff
@@ -20,16 +20,17 @@
 source $(dirname $0)/../vendor/knative.dev/hack/release.sh
 
 function build_release() {
-  # Run `generate-yamls.sh`, which should be versioned with the
-  # branch since the detail of building may change over time.
-  local YAML_LIST="$(mktemp)"
-  export TAG
-  $(dirname $0)/generate-yamls.sh "${REPO_ROOT_DIR}" "${YAML_LIST}"
-  ARTIFACTS_TO_PUBLISH=$(cat "${YAML_LIST}" | tr '\n' ' ')
-  if (( ! PUBLISH_RELEASE )); then
-    # Copy the generated YAML files to the repo root dir if not publishing.
-    cp ${ARTIFACTS_TO_PUBLISH} ${REPO_ROOT_DIR}
-  fi
+	# Run `generate-yamls.sh`, which should be versioned with the
+	# branch since the detail of building may change over time.
+	local YAML_LIST
+	YAML_LIST="$(mktemp)"
+	export TAG
+	$(dirname $0)/generate-yamls.sh "${REPO_ROOT_DIR}" "${YAML_LIST}"
+	ARTIFACTS_TO_PUBLISH=$(tr '\n' ' ' <"${YAML_LIST}")
+	if ((!PUBLISH_RELEASE)); then
+		# Copy the generated YAML files to the repo root dir if not publishing.
+		cp "${ARTIFACTS_TO_PUBLISH}" "${REPO_ROOT_DIR}"
+	fi
 }
 
-main $@
+main "$@"
```

Comment:
This change looks good to me, but I just need a second pair of eyes on this as this is rather important script.

Reply:
The issue with the release script is, it is only run when there's a new release. In the case of other scripts, the CI runs the script and we would see any regression very quickly.

Reply:
Ah no problem, I was looking at it and shellcheck prompted with some diagnostics/suggestions to improve this, that namely being using quotes "" on variables to prevent globbing and to direct input from the variable directly instead of using cat

Reply:
This will fail. In fact, this script uses globing to copy those paths, which is bad, but you can't just blindly apply shellcheck hints. 

- Line 32 will fail, if there is more than one YAML (there is).
- IDK about line 30. It looks sketchy.

Reply:
Happy to merge the PR if you want to skip the changes in this specific file @amaanq